Feature/bdue h qo9 6 item details page

an-nguyen-jh opened the pull request.

an-nguyen-jh committed:
[Fix] Modify Appbar to use in admin page

an-nguyen-jh committed:
[Style] Create toolbar containing change view and sort

Create Toolbar component containing sort options and change layout view

Add extend className for Select component

Rename formData to options

an-nguyen-jh committed:
[Style] Create list view layout and handle responsive

an-nguyen-jh committed:
[Style] Set default Image when no image available

an-nguyen-jh committed:
[Style] Create grid view and create device info card

an-nguyen-jh committed:
[Fix] Fix device info card not center in mobile view and wrong media query for tablet

an-nguyen-jh committed:
[Fix] Change default value of DeviceInfo

an-nguyen-jh committed:
[Feat] Get, sort device info and display in ListView

an-nguyen-jh committed:
[Feat] Display device info in grid view

an-nguyen-jh committed:
[Fix] Change logic of handle sort and paginate in deviceInfo

an-nguyen-jh committed:
[Feat] Handle pagination in Device Info List

an-nguyen-jh committed:
[Style] Add delete button in List view and Grid view

an-nguyen-jh committed:
[Feat] Create confirm delete device info dialog

an-nguyen-jh committed:
[Style] Show info in confirm delete dialog

an-nguyen-jh committed:
[Feat] Handle delete device info and relative deivce request

an-nguyen-jh committed:
[Feat] Delete image of device info and reload device info list

an-nguyen-jh committed:
[Fix] Move, rename function and clear console log

anh-dinh-jh commented on src/components/Dialog/ConFirmDeleteDialog.js:
```diff
@@ -0,0 +1,90 @@
+import React from "react";
```
Incorrect naming

anh-dinh-jh commented on src/components/Dialog/ConFirmDeleteDialog.js:
```diff
@@ -0,0 +1,90 @@
+
+  const handleDeleteDeviceInfo = async () => {
+    try {
+      await deleteDeviceInfoByEmail(email);
+      await deleteAllRelativeDeviceRequests(email);
+      await deleteOldEmployeeDeviceImages(imageSrcs);
```
If they don't depend on each other, can we just call them concurrently instead of calling one by one?

anh-dinh-jh commented on src/components/EmployeeDevice/EmployeeDevice.js:
```diff
@@ -0,0 +1,124 @@
+        const deviceInfoSnapshots = await getAllDeviceInfos();
+        const tempDeviceInfos = [];
+        deviceInfoSnapshots.forEach((deviceInfoSnap) => {
+          const temp = deviceInfoSnap.data();
```
Should not use the unclear name

anh-dinh-jh commented on src/components/Pagination/Pagination.js:
```diff
@@ -0,0 +1,43 @@
+  return (
+    <div className="container-fluid" style={{ display: "flex" }}>
+      <ul className="pagination">
+        {[2, 1].map(function generatePrevPage(index) {
```
Should use a better way to get the array for pagination

anh-dinh-jh commented on src/components/Pagination/Pagination.js:
```diff
@@ -0,0 +1,43 @@
+        <li className="pagination__link--active pagination__link">
+          {currentPage}
+        </li>
+        {[1, 2].map(function generatePrevPage(index) {
```
Same as above

anh-dinh-jh commented on src/utils/sort.js:
```diff
@@ -0,0 +1,24 @@
+export function sortDeviceInfos(deviceInfos, sortTokens) {
+  if (sortTokens[0] === "name") {
+    if (sortTokens[1] === "desc") {
+      deviceInfos.sort((a, b) => {
+        return b.name.localeCompare(a.name);
+      });
+    } else {
+      deviceInfos.sort((a, b) => {
+        return a.name.localeCompare(b.name);
+      });
+    }
```
Can we shorten this?  Or at least we can have a general function for sorting that can be used anywhere else.

anh-dinh-jh commented on src/utils/sort.js:
```diff
@@ -0,0 +1,24 @@
+      });
+    }
+  }
+  if (sortTokens[0] === "updatedTime") {
+    if (sortTokens[1] === "desc") {
+      deviceInfos.sort((a, b) => {
+        return b.updatedTime > a.updatedTime ? 1 : -1;
+      });
+    } else {
+      deviceInfos.sort((a, b) => {
+        return a.updatedTime > b.updatedTime ? 1 : -1;
+      });
```
Same as above

an-nguyen-jh committed:
[Fix] Fix mistake found in code review

Rename file ConFirmDeleteDialog

Implement another ways to create Pagination

Using Promise.all to handle multiple async functions

Shorten sort function by using prototype

anh-dinh-jh commented on src/components/Pagination/Pagination.js:
```diff
@@ -1,38 +1,32 @@
 function Pagination({ currentPage, handlePageChange, totalItem, limit }) {
+  const pageIndexes = [currentPage];
+
+  for (let i = 1; i <= 2; i++) {
```
What does "2" stand for?

an-nguyen-jh replied:
It stand for next 2 page for previous 2 page from current page. Should i use constant for it ?

anh-dinh-jh commented on src/utils/sort.js:
```diff
@@ -1,24 +1,24 @@
+};
+
+export function sortByElementProperty(unsortedArray, sortTokens) {
+  if (sortTokens.length === 2) {
```
What does "2" stand for?

an-nguyen-jh replied:
it prevents error when sortTokens doesn't have anything token to select sort prototype.

an-nguyen-jh committed:
[Fix] Define constant for pagination range and sortTokens length

anh-dinh-jh commented on src/components/Pagination/Pagination.js:
```diff
@@ -4,8 +4,9 @@ import classNames from "classnames";
 
 function Pagination({ currentPage, handlePageChange, totalItem, limit }) {
   const pageIndexes = [currentPage];
+  const paginationRange = 2;
```
Can we get the total of pages by the total of items?

an-nguyen-jh replied:
Actually, we can, but i do this for the limit of page indexes. if we have a total of 6 pages and our current page is 4 so the page indexes are 2,3,4,5,6

anh-dinh-jh replied:
We should make sure that it always works without modifying in the future just to extend the number of pages. For your case, we can have the next & last buttons or re-do the pagination.

an-nguyen-jh committed:
[Fix] re-style for heading and paragraph in dialog
